src/taskgraph/transforms/from_deps.py

```diff
@@ -41,12 +70,14 @@
                 "set-name",
                 description=dedent(
                     """
-                When True, `from_deps` will derive a name for the generated
-                tasks from the name of the primary dependency. Defaults to
-                True.
+                UPDATE ME AND DOCS
                 """.lstrip()
                 ),
-            ): bool,
+            ): Any(
+                None,
+                *SET_NAME_MAP,
+                {Any(*SET_NAME_MAP): object},
```

Comment:
This is going to break the ability to create custom `set_name` functions because the Schema is created at import time which happens before said custom function will have a change to register themselves. That's why I ultimately moved the `group_by` function off into a utility file. That way we have a chance to register functions before the schema gets created.

Alternatively we could just allow any string rather than the actually registered ones. That would be fine with me.

Reply:
Ah, good point. It's pretty easy to just move this out elsewhere, and matches what we're doing for other things. Let's do that.

I am curious though: is there a reason not to just accept a callable here instead?

Reply:
Like the python path notation? Yeah that would work too.

For `group-by` I was just copy/pasting how the existing `multi_dep` loader did it. One benefit of this way is that consumers don't need to care about where the function lives, e.g `set-name: default` vs `set-name: taskgraph.transforms.from_deps.default`. It looks cleaner.